File: src/p0014/cpp/solution.cpp
class Solution {
public:
    string longestCommonPrefix(const vector<string> &v) {
        string result;
        if (v.empty()) return result;
        for (size_t i = 0;; i++) {
            for (size_t j = 0; j < v.size(); j++) {
                if (i >= v[j].size() || (v[j][i] != v[0][i])) {
                    return result;
                }
            }
            result.push_back(v[0][i]);
        }
    }
};

File: src/p0014/python/solution.py
from typing import List


class Solution:
    def longestCommonPrefix(self, v: List[str]) -> str:
        if len(v) == 0:
            return ""
        result, i = "", 0
        while True:
            for j in range(0, len(v)):
                if i >= len(v[j]) or v[j][i] != v[0][i]:
                    return result
            result += v[0][i]
            i = i + 1
